### PR TITLE
fix(workitem): Do not set the `ID` for the work item to create in the 'quick add' component (#1948)

### PR DESCRIPTION
**Pull request by xcoulon**
Removed the unwanted line that set the `ID` to '42' in the workitem creation request.

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1949-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1949-fabric8-planner.int.rdu2c.fabric8.io

File: src/app/components/work-item-quick-add/work-item-quick-add.component.ts [hunk 1]
```diff
@@ -105,7 +105,6 @@ export class WorkItemQuickAddComponent implements OnInit, OnDestroy, OnChanges,
     this.workItem.attributes = new Map<string, string | number>();
     this.workItem.relationships = new WorkItemRelations();
     this.workItem.type = 'workitems';
-    this.workItem.id = '42';
```
Comment (joshuawilson) [inline src/app/components/work-item-quick-add/work-item-quick-add.component.ts]: Is this set in the backend? Should it be set to Null?

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1949-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1949-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1949-fabric8-planner.int.rdu2c.fabric8.io

File: src/app/components/work-item-quick-add/work-item-quick-add.component.ts (hunk 1 above)
Comment (pranavgore09) [inline src/app/components/work-item-quick-add/work-item-quick-add.component.ts]: @nimishamukherjee @joshuawilson 
1. If ID is not set
2. If ID is set to `null`
In above both cases, API creates a new work item and returns a valid response.

Comment (sanbornsen) [inline src/app/components/work-item-quick-add/work-item-quick-add.component.ts]: Long long time ago, ID null or no ID used to give error from back end :P

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1949-fabric8-planner.int.rdu2c.fabric8.io

File: src/app/components/work-item-quick-add/work-item-quick-add.component.ts (hunk 1 above)
Comment (xcoulon) [inline src/app/components/work-item-quick-add/work-item-quick-add.component.ts]: the problem here is that `42` is not a valid `uuid` and furthermore, it does not make sense to provide a value for this field in the creation request. I can understand it was a work-around for an old bug, but it should not be kept any longer.

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1949-fabric8-planner.int.rdu2c.fabric8.io

**Pull request merged**